app/controllers/api/v1/comments_controller.rb

```diff
@@ -0,0 +1,67 @@
+class Api::V1::CommentsController < ApplicationController
+  before_action :set_comment, only: %i[ show update destroy ]
+  include Authentication
+  before_action :authenticate
+
+  # GET /comments
+  def index
+    comments = Comment.where(post_id: params[:post_id])
+
+    render json: comments
+  end
+
+  # GET /comments/1
+  def show
+    render json: @comment 
+  end
+
+  # POST /comments
+  def create
+    post = Post.find(params[:post_id])
+    comment = Comment.new(post: post, user: @user, body: comment_params[:body])
+
+    if comment.save 
+      comments_count = post.comments_count + 1
+      post.update(comments_count: comments_count)
+
+      render json: {data: comment}, status: 200
+    else
+      render status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /comments/1
+  def update
+    if @comment.user_id == @user.id
```

Comment:
this can also be centralized as Authorization, as it will be reused across multiple actions naturally

Reply:
Got it. Will centralize it.